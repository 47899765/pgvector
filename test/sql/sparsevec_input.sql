SELECT '{0:1.5,2:3.5}'::sparsevec;
SELECT '{0:1.5,2:3.5}'::sparsevec::vector;
SELECT '{0:1.5,2:3.5}'::sparsevec::vector(5);
SELECT '{0:1.5,2:3.5}'::sparsevec::vector(4);
SELECT '{0:1.5,2:3.5}'::sparsevec::vector(2);
SELECT '[0,1.5,0,3.5,0]'::vector::sparsevec;

SELECT '{0:0,1:1,2:0}'::sparsevec;

SELECT '{1:1,0:1}'::sparsevec;

SELECT '{}'::sparsevec;
SELECT '{}'::sparsevec::vector;

SELECT '{-1:1}'::sparsevec;
SELECT '{1:1}'::sparsevec;

SELECT '{}'::sparsevec(2);
