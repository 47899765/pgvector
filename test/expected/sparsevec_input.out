SELECT '{0:1.5,2:3.5}'::sparsevec;
   sparsevec   
---------------
 {0:1.5,2:3.5}
(1 row)

SELECT '{0:1.5,2:3.5}'::sparsevec::vector;
   vector    
-------------
 [1.5,0,3.5]
(1 row)

SELECT '{0:1.5,2:3.5}'::sparsevec::vector(5);
     vector      
-----------------
 [1.5,0,3.5,0,0]
(1 row)

SELECT '{0:1.5,2:3.5}'::sparsevec::vector(4);
    vector     
---------------
 [1.5,0,3.5,0]
(1 row)

SELECT '{0:1.5,2:3.5}'::sparsevec::vector(2);
ERROR:  Vector must have at least 3 dimensions
SELECT '[0,1.5,0,3.5,0]'::vector::sparsevec;
   sparsevec   
---------------
 {1:1.5,3:3.5}
(1 row)

SELECT '{0:0,1:1,2:0}'::sparsevec;
 sparsevec 
-----------
 {1:1}
(1 row)

SELECT '{1:1,0:1}'::sparsevec;
ERROR:  indexes must be in ascending order
LINE 1: SELECT '{1:1,0:1}'::sparsevec;
               ^
SELECT '{}'::sparsevec;
 sparsevec 
-----------
 {}
(1 row)

SELECT '{}'::sparsevec::vector;
ERROR:  vector must have at least 1 dimension
SELECT '{-1:1}'::sparsevec;
ERROR:  index "-1" is out of range for type sparsevec
LINE 1: SELECT '{-1:1}'::sparsevec;
               ^
SELECT '{1:1}'::sparsevec;
 sparsevec 
-----------
 {1:1}
(1 row)

SELECT '{}'::sparsevec(2);
ERROR:  type modifier is not allowed for type "sparsevec"
LINE 1: SELECT '{}'::sparsevec(2);
                     ^
